tests/unit/stream_alert_rule_processor/test_parsers.py

```diff
@@ -90,6 +92,124 @@ class TestJSONParser(TestParser):
     def _parser_type(cls):
         return 'json'
 
+    def test_schema_fail(self):
+        """JSON Parser - Schema Match Failure"""
+        # setup
+        schema = self.config['logs']['unit_test_simple_log']['schema']
+        data = json.dumps({'hey': 1})
+
+        # get parsed data
+        parsed_data = self.parser_helper(data=data, schema=schema)
+
+        assert_false(parsed_data)
+
+    def test_non_string_input(self):
+        """JSON Parser - Non String Input"""
+        # setup
+        schema = {'name': 'string', 'result': 'string'}
+        data = {'name': 'test', 'result': 'test'}
+
+        # get parsed data
+        parsed_data = self.parser_helper(data=data, schema=schema)
+
+        assert_equal(len(parsed_data), 1)
+
+    @patch('stream_alert.rule_processor.parsers.LOGGER')
+    def test_invalid_json(self, mock_logging):
+        """JSON Parser - Invalid Input"""
+        # setup
+        schema = {'name': 'string', 'result': 'string'}
+        data = 'not json data'
+
+        # get parsed data
+        parsed_data = self.parser_helper(data=data, schema=schema)
+
+        assert_true(mock_logging.debug.called)
+        assert_false(parsed_data)
+
+    def test_envelope_keys_optional_values(self):
+        """JSON Parser - Default Values for Missing Envelope Keys"""
+        options = self.config['logs']['json:regex_key_with_envelope']['configuration']
+        schema = self.config['logs']['json:regex_key_with_envelope']['schema']
+        # missing 'host' envelope key
+        data = json.dumps({
+            'message': '{"nested_key_1": "test"'
+                       ', "nested_key_2": "test"'
+                       ', "nested_key_3": "test"}',
+            'date': '2017/10/01',
+            'time': '14:00:00'
+        })
+
+        # get parsed data
+        parsed_data = self.parser_helper(data=data, schema=schema, options=options)
+
+        assert_equal(len(parsed_data), 1)
+        assert_true(parsed_data[0]['streamalert:envelope_keys']['host'] == '')
+
+    @patch('stream_alert.rule_processor.parsers.LOGGER')
+    def test_regex_key_invalid_json(self, mock_logging):
+        """JSON Parser - Regex Key with Invalid JSON Object"""
+        options = self.config['logs']['json:regex_key_with_envelope']['configuration']
+        schema = self.config['logs']['json:regex_key_with_envelope']['schema']
+        # Invalid JSON and missing 'host' envelope key
+        data = json.dumps({
+            'message': '{"nested_key_1": "test"'
+                       ' "nested_key_2": "test"'
+                       ' "nested_key_3": "test"}',
+            'date': '2017/10/01',
+            'time': '14:00:00'
+        })
+
+        # get parsed data
+        parsed_data = self.parser_helper(data=data, schema=schema, options=options)
+
+        assert_true(mock_logging.debug.called)
+        assert_false(parsed_data)
+
+    def test_regex_key_non_json(self):
+        """JSON Parser - Regex Key with Invalid JSON Object"""
```

Comment:
I just noticed this test has the same description as the one above it. Can you update one of them please?